the typo fixes and comment additions look fine, but this PR does not resolve all lint errors:

```
Errors from golint:
staging/src/k8s.io/kubectl/pkg/cmd/create/create.go:50:6: type name will be used as create.CreateOptions by other packages, and that stutters; consider calling this Options
staging/src/k8s.io/kubectl/pkg/cmd/create/create.go:342:6: type name will be used as create.CreateSubcommandOptions by other packages, and that stutters; consider calling this SubcommandOptions
staging/src/k8s.io/kubectl/pkg/cmd/create/create_clusterrole.go:65:6: type name will be used as create.CreateClusterRoleOptions by other packages, and that stutters; consider calling this ClusterRoleOptions
staging/src/k8s.io/kubectl/pkg/cmd/create/create_cronjob.go:53:6: type name will be used as create.CreateCronJobOptions by other packages, and that stutters; consider calling this CronJobOptions
staging/src/k8s.io/kubectl/pkg/cmd/create/create_deployment.go:60:6: type name will be used as create.CreateDeploymentOptions by other packages, and that stutters; consider calling this DeploymentOptions
staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go:105:6: type name will be used as create.CreateIngressOptions by other packages, and that stutters; consider calling this IngressOptions
staging/src/k8s.io/kubectl/pkg/cmd/create/create_job.go:56:6: type name will be used as create.CreateJobOptions by other packages, and that stutters; consider calling this JobOptions
staging/src/k8s.io/kubectl/pkg/cmd/create/create_role.go:123:6: type name will be used as create.CreateRoleOptions by other packages, and that stutters; consider calling this RoleOptions
```

we've declined to rename these in the past just to satisfy lint; I would suggest restoring the lint exception for this package

--- a/staging/src/k8s.io/kubectl/pkg/cmd/create/create_cronjob.go
+++ b/staging/src/k8s.io/kubectl/pkg/cmd/create/create_cronjob.go
@@ -49,6 +49,7 @@ var (
 		kubectl create cronjob my-job --image=busybox --schedule="*/1 * * * *" -- date`)
 )
 
+// CreateCronJobOptions is returned by NewCreateCronJobOptions
 type CreateCronJobOptions struct {
 	PrintFlags *genericclioptions.PrintFlags
 
@@ -72,6 +73,7 @@ type CreateCronJobOptions struct {
 	genericclioptions.IOStreams
 }
 
+// NewCreateCronJobOptions returns an initialized CreateCronJobOptions instance
 func NewCreateCronJobOptions(ioStreams genericclioptions.IOStreams) *CreateCronJobOptions {
 	return &CreateCronJobOptions{
 		PrintFlags: genericclioptions.NewPrintFlags("created").WithTypeSetter(scheme.Scheme),
@@ -110,6 +112,7 @@ func NewCmdCreateCronJob(f cmdutil.Factory, ioStreams genericclioptions.IOStream
 	return cmd
 }
 
+// Complete completes all the required options
 func (o *CreateCronJobOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
 	name, err := NameFromCommandArgs(cmd, args)
 	if err != nil {
@@ -165,6 +168,7 @@ func (o *CreateCronJobOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, a
 	return nil
 }
 
+// Run performs the execution of 'create cronjob' sub command
 func (o *CreateCronJobOptions) Run() error {
 	cronjob := o.createCronJob()
 

--- a/staging/src/k8s.io/kubectl/pkg/cmd/create/create_deployment.go
+++ b/staging/src/k8s.io/kubectl/pkg/cmd/create/create_deployment.go
@@ -79,6 +79,7 @@ type CreateDeploymentOptions struct {
 	genericclioptions.IOStreams
 }
 
+// NewCreateDeploymentOptions returns an initialized CreateDeploymentOptions instance
 func NewCreateDeploymentOptions(ioStreams genericclioptions.IOStreams) *CreateDeploymentOptions {
 	return &CreateDeploymentOptions{
 		Port:       -1,
@@ -173,6 +174,7 @@ func (o *CreateDeploymentOptions) Complete(f cmdutil.Factory, cmd *cobra.Command
 	return nil
 }
 
+// Validate makes sure there is no discrepency in provided option values
 func (o *CreateDeploymentOptions) Validate() error {
 	if len(o.Images) > 1 && len(o.Command) > 0 {
 		return fmt.Errorf("cannot specify multiple --image options and command")

--- a/staging/src/k8s.io/kubectl/pkg/cmd/create/create_priorityclass.go
+++ b/staging/src/k8s.io/kubectl/pkg/cmd/create/create_priorityclass.go
@@ -71,6 +71,7 @@ type PriorityClassOptions struct {
 	genericclioptions.IOStreams
 }
 
+// NewPriorityClassOptions returns an initialized PriorityClassOptions instance
 func NewPriorityClassOptions(ioStreams genericclioptions.IOStreams) *PriorityClassOptions {
 	return &PriorityClassOptions{
 		Value:            0,

--- a/staging/src/k8s.io/kubectl/pkg/cmd/create/create_rolebinding.go
+++ b/staging/src/k8s.io/kubectl/pkg/cmd/create/create_rolebinding.go
@@ -45,7 +45,7 @@ var (
 		  kubectl create rolebinding admin --clusterrole=admin --user=user1 --user=user2 --group=group1`))
 )
 
-// RoleBindingOpts holds the options for 'create rolebinding' sub command
+// RoleBindingOptions holds the options for 'create rolebinding' sub command
 type RoleBindingOptions struct {
 	PrintFlags *genericclioptions.PrintFlags
 	PrintObj   func(obj runtime.Object) error

